### PR TITLE
refactor: Add bitcoin derivation path guard

### PR DESCRIPTION
**Pull request by ujjwal-cyph**
Implements https://app.clickup.com/t/9002019994/PRF-4825

Add specialized API for validating derivation paths for the Bitcoin chain only. The existing API for this purpose has mixed handling for all the Bitcoin forks and Ethereum as well.

### CODE REVIEW

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-4825 Analyze bitcoin forks handling](https://app.clickup.com/t/9002019994/PRF-4825)

File: apps/btc_app/get_public_key.c [hunk 1]
```diff
@@ -167,8 +168,8 @@ static bool validate_request_data(btc_get_public_key_request_t *request) {
 
   // TODO: Enable btc/coin specific check
   if (false ==
-      verify_receive_derivation_path(request->initiate.derivation_path,
-                                     request->initiate.derivation_path_count)) {
+      btc_derivation_path_guard(request->initiate.derivation_path,
```
Comment (kartikCypherock) [inline apps/btc_app/get_public_key.c:171]:
```suggestion
      !btc_derivation_path_guard(request->initiate.derivation_path,
```

File: apps/btc_app/get_xpub.c [hunk 2]
```diff
@@ -196,7 +197,7 @@ static bool validate_request_data(btc_get_xpubs_request_t *request) {
   for (pb_size_t index = 0; index < count; index++) {
     path = &request->initiate.derivation_paths[index];
     // TODO: Enable btc/coin specific check
-    if (false == verify_xpub_derivation_path(path->path, path->path_count)) {
+    if (false == btc_derivation_path_guard(path->path, path->path_count)) {
```
Comment (kartikCypherock) [inline apps/btc_app/get_xpub.c:200]:
```suggestion
    if (!btc_derivation_path_guard(path->path, path->path_count)) {
```

**Review by kartikCypherock — APPROVED** [review]

File: apps/btc_app/get_public_key.c [hunk 3]
```diff
@@ -167,8 +168,8 @@ static bool validate_request_data(btc_get_public_key_request_t *request) {
...
+                                request->initiate.derivation_path_count)) {
```
Comment (ujjwal-cyph) [inline apps/btc_app/get_public_key.c:172]:
```suggestion
  if (!btc_derivation_path_guard(request->initiate.derivation_path,
                                 request->initiate.derivation_path_count)) {
```

File: apps/btc_app/get_public_key.c [hunk 4]
```diff
@@ -167,8 +168,8 @@ static bool validate_request_data(btc_get_public_key_request_t *request) {
 
   // TODO: Enable btc/coin specific check
   if (false ==
```
Comment (ujjwal-cyph) [inline apps/btc_app/get_public_key.c:170]:
```suggestion
```

**Review by amanCypherock — APPROVED** [review]

**Pull request merged**